analyzer/src/main/kotlin/managers/Bower.kt

```diff
@@ -141,13 +141,69 @@ class Bower(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfigu
             return result
         }
 
+        private fun hasCompleteDependencies(node: JsonNode, scopeName: String): Boolean {
+            val dependencyKeys = node["dependencies"].fieldNamesOrEmpty().asSequence().toSet()
+            val dependencyRefKeys = node["pkgMeta"][scopeName].fieldNamesOrEmpty().asSequence().toSet()
+
+            return dependencyKeys.containsAll(dependencyRefKeys)
+        }
+
+        private fun dependencyKeyOf(node: JsonNode): String? {
+            // As non-null dependency keys are supposed to as equivalence relation for parsing 'missing' nodes,
+            // thus only the name and version attributes can be used. Typically those attributes should be not null
+            // however in particular for root projects the null case also happens.
+            val name = node["pkgMeta"]["name"].textValueOrEmpty()
+            val version = node["pkgMeta"]["version"].textValueOrEmpty()
+            return if (name.isEmpty() || version.isEmpty()) null else "$name:$version"
```

Comment:
Is it guaranteed that the `pkgMeta` field includes a `version` attribute? I don't remember whether this JSON represents the declared dependencies or the packages that were installed as a result of dependency resolution.

Reply:
To my understanding it is not guaranteed that this version field is included. In particular for our synthetic bower project this is not the case for the root of the dep tree - and this parses fine through Bower.